client/web/src/components/popovers/ColorPicker.vue

```diff
@@ -115,13 +117,197 @@
 
 <script lang="ts">
 import { defineComponent } from "vue";
+import { clamp, HSV2RGB, RGB2HSV, RGB, HSV } from "../../lib/utils";
+
+interface Rect {
+	width: number;
+	height: number;
+	left: number;
+	top: number;
+}
+
+interface ColorPickerData {
+	color: HSV;
+	hue: {
+		rect: Rect;
+	};
+	saturation: {
+		rect: Rect;
+	};
+	opacity: {
+		rect: Rect;
+	};
+}
+
+type ColorPickerState = "idle" | "move_hue" | "move_opacity" | "move_saturation";
```

Comment:
An enum would be preferred here since it fits the pattern used in the existing codebase.